### PR TITLE
Normalize Grafana setup

### PR DESCRIPTION
**Pull request by brancz**
This normalizes the Grafana setup, so Grafana does not depend on any
"default" paths, but instead the only interface for the container image
is that the entrypoint execs a Grafana server binary.

@mxinden @squat

### CODE REVIEW

File: hack/ocp-images.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    ose-configmap-reloader
+    ose-cluster-monitoring-operator
+    ose-kube-state-metrics
+    ose-kube-rbac-proxy
+    ose-prometheus-config-reloader
+    ose-prometheus-operator
+)
+
+for i in ${IMAGES[@]}; do
+    docker pull $SRC_REGISTRY/$i:$TAG
```
Comment (mxinden) [inline hack/ocp-images.sh]: `s/$SRC_REGISTRY/$i:$TAG/"{$SRC_REGISTRY}/${i}:${TAG}"`

File: hack/ocp-images.sh [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        name: cluster-monitoring-operator
+        args:
+        - "-namespace=openshift-monitoring"
+        - "-configmap=cluster-monitoring-config"
+        - "-logtostderr=true"
+        - "-v=4"
+        - "-tags=prometheus-operator=$TAG"
+        - "-tags=prometheus-config-reloader=$TAG"
+        - "-tags=config-reloader=$TAG"
+        #- "-tags=prometheus=$TAG"
```
Comment (squat) [inline hack/ocp-images.sh]: Why do we need these commented out args?

File: hack/ocp-images.sh [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    ose-prometheus-config-reloader
+    ose-prometheus-operator
+)
+
+for i in ${IMAGES[@]}; do
+    docker pull $SRC_REGISTRY/$i:$TAG
+done
+
+for i in ${IMAGES[@]}; do
+    docker tag $SRC_REGISTRY/$i:$TAG $DST_REGISTRY/$i:$TAG
```
Comment (mxinden) [inline hack/ocp-images.sh]: See above.

File: hack/ocp-images.sh [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+done
+
+for i in ${IMAGES[@]}; do
+    docker push $DST_REGISTRY/$i:$TAG
```
Comment (mxinden) [inline hack/ocp-images.sh]: See above.

File: hack/ocp-images.sh [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+set -e
+set -x
+# only exit with zero if all commands of the pipeline exit successfully
+set -o pipefail
+
+# This script synchronizes OCP images into a cluster-local registry of an
+# OpenShift cluster, and writes the appropriate configuration into the
+# cluster-monitoring-operator manifests in `manifests/`.
+
+SRC_REGISTRY=registry.reg-aws.openshift.com:443/openshift3
```
Comment (squat) [inline hack/ocp-images.sh]: do we need to specify 443 here? and why not on `DST_REGISTRY`?

File: hack/ocp-images.sh [hunk 6]
```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
+set -e
+set -x
+# only exit with zero if all commands of the pipeline exit successfully
+set -o pipefail
+
+# This script synchronizes OCP images into a cluster-local registry of an
+# OpenShift cluster, and writes the appropriate configuration into the
+# cluster-monitoring-operator manifests in `manifests/`.
```
Comment (mxinden) [inline hack/ocp-images.sh]: Maybe worth mentioning that it expects the `TAG` environment variable.

File: hack/ocp-images.sh (hunk 2 above)
Comment (brancz) [reply hack/ocp-images.sh]: I intentionally did this, as this script syncs all the other images except prometheus, alertmanager, node-exporter and oauth-proxy. But I wanted to leave these here for documentation purposes.

File: hack/ocp-images.sh (hunk 5 above)
Comment (brancz) [reply hack/ocp-images.sh]: Good point, let me check if that makes a difference.

Comment (brancz) [reply hack/ocp-images.sh]: It is not. Updating.

**Pull request merged**